src/test/run-pass/shift-near-oflo.rs

```diff
@@ -13,9 +13,6 @@
 // Check that we do *not* overflow on a number of edge cases.
 // (compare with test/run-fail/overflowing-{lsh,rsh}*.rs)
 
-// (Work around constant-evaluation)
-fn id<T>(x: T) -> T { x }
```

Comment:
Yes, I was overzealous here.

The problem is, that this test gets *silently* turned into a compile-time test whenever we improve the const evaluator. I think if we made the `id` function in a way that it modifies a `static` on every call, there's no way this can ever be evaluated at compile-time in rustc. LLVM will probably still end up optimizing everything out. So this test also needs to guarantee that optimizations are off (are they even on in tests? if they are not, ever activating them will again silently turn this test into a compile-time test).

An alternative is to add a text file from which the values are read. This way optimizations and the const evaluator cannot ever influence the meaning of this test.